File: ruby/lib/steg_tools/steg_automate.rb
require 'safe_shell'
require_relative 'steg_tools'

class StegTools::StegAutomate
  attr_accessor :search_strings, :options
  attr_reader :strings_result

  def initialize(options)
    @options = options
  end

  def search_strings
    @strings_result = SafeShell.execute('strings', @options[:file])
    unless $?.success?
      raise SafeShell::CommandFailedException.new("Shell command #{['strings', @options[:file]].inspect} failed with status #{$?}")
    end
  rescue => err
    case err
    when Errno::ENOENT then puts "Error: File not found."
    when SafeShell::CommandFailedException then puts err.message
    else puts "Error: Something gone wrong."
    end
    exit
  end

  def file_contain_string?
    @strings_result.include?(@options[:text_to_search])
  end

  def index_of_pattern
    @strings_result.index(@options[:text_to_search])
  end

  def index_of_delimetter(start)
    new_string = @strings_result[start..-1]
    return new_string.index(@options[:pattern]) + start
  end

  def search_for_specific_string
    start = index_of_pattern
    if file_contain_string?
      @strings_result = @strings_result[start..index_of_delimetter(start)]
    end
  end

  def run
    search_strings
    search_for_specific_string
    puts "[x] flag not found !!!" if @strings_result.nil?
    puts "[+] the flag is : #{@strings_result}" unless @strings_result.nil?
  end
end

File: python/steg_tools/steg_automate.py
import subprocess

class StegAutomate:
    def __init__(self, options):
        self.options = options
        self.strings_result = None

    def search_strings(self):
        self.strings_result = str(subprocess.check_output(['strings ' + self.options['file']], stderr=subprocess.STDOUT, shell=True))

    def file_contains_string(self):
        return self.options['text_to_search'] in self.strings_result

    def index_of_pattern(self):
        return self.strings_result.find(self.options['text_to_search'])

    def index_of_delimiter(self, start):
        new_string = self.strings_result[start:]
        return new_string.find(self.options['pattern']) + start + 1

    def search_for_specific_string(self):
        start = self.index_of_pattern()
        if self.file_contains_string():
            return self.strings_result[start:self.index_of_delimiter(start)]

    def run(self):
        self.search_strings()
        result = self.search_for_specific_string()
        print("[x] flag not found !!!" if result is None else f"[+] the flag is : {result}")